pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3011,8 +3012,37 @@ int BPF_KPROBE(trace_security_bpf)
     if (should_submit(SECURITY_BPF, &(p.event->context))) {
         // 1st argument == cmd (int)
         save_to_submit_buf(p.event, (void *) &cmd, sizeof(int), 0);
-
         events_perf_submit(&p, SECURITY_BPF, 0);
+
+        // Capture BPF object loaded
+        if (cmd == BPF_PROG_LOAD && p.config->options & OPT_CAPTURE_BPF) {
+            bin_args_t bin_args = {};
+            u32 pid = p.task_info->context.host_pid;
+
+            union bpf_attr *attr = (union bpf_attr *) PT_REGS_PARM2(ctx);
+            u32 insn_cnt = get_attr_insn_cnt(attr);
+            const struct bpf_insn *insns = get_attr_insns(attr);
+            unsigned int insn_size = (unsigned int) (sizeof(struct bpf_insn) * insn_cnt);
+
+            bin_args.type = SEND_BPF_OBJECT;
+            char prog_name[16] = {0};
+            long sz = bpf_probe_read_str(prog_name, 16, attr->prog_name);
+            if (sz > 0) {
+                sz = bpf_probe_read_str(bin_args.metadata, sz, prog_name);
+            }
+
+            u32 rand = bpf_get_prandom_u32();
+            bpf_probe_read(&bin_args.metadata[16], 4, &rand);
+            bpf_probe_read(&bin_args.metadata[20], 4, &pid);
+            bpf_probe_read(&bin_args.metadata[24], 4, &insn_size);
+            bin_args.ptr = (char *) insns;
+            bin_args.start_off = 0;
+            bin_args.full_size = insn_size;
+
+            u64 id = bpf_get_current_pid_tgid();
+            bpf_map_update_elem(&bin_args_map, &id, &bin_args, BPF_ANY);
+            bpf_tail_call(ctx, &prog_array, TAIL_SEND_BIN);
```

Comment:
The tail call here means we skip over the following lines:
```
union bpf_attr *attr = (union bpf_attr *) PT_REGS_PARM2(ctx);

reset_event_args(&p);
check_bpf_link(&p, attr, cmd);
```

Are you sure this is ok?

Reply:
fixed, thanks